src/creature_instances.c

```diff
@@ -494,19 +496,28 @@ long instf_creature_cast_spell(struct Thing *creatng, long *param)
     struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
     long spl_idx = param[0];
     struct SpellConfig* spconf = get_spell_config(spl_idx);
-    SYNCDBG(8,"The %s index %d casts %s",thing_model_name(creatng),(int)creatng->index,spell_code_name(spl_idx));
-    if (spconf->cast_at_thing)
+    struct Thing* target = NULL;
+
+    SYNCDBG(8,"The %s(%d) casts %s at %d", thing_model_name(creatng), (int)creatng->index,
+        spell_code_name(spl_idx), cctrl->targtng_idx);
+
+    if (spconf->cast_at_thing && cctrl->targtng_idx != creatng->index)
     {
-        struct Thing* trthing = thing_get(cctrl->targtng_idx);
-        if (!thing_is_invalid(trthing))
-        {
-            creature_cast_spell_at_thing(creatng, trthing, spl_idx, cctrl->explevel);
-            // Start cooldown after spell is cast
-            cctrl->instance_use_turn[cctrl->instance_id] = game.play_gameturn;
-            return 0;
-        }
+        // If the targtng_idx is jus the caster itself, we can call creature_cast_spell
+        // instead of creature_cast_spell_at_thing.
+        target = thing_get(cctrl->targtng_idx);
+        if (thing_is_invalid(target)) target = NULL;
     }
-    creature_cast_spell(creatng, spl_idx, cctrl->explevel, cctrl->targtstl_x, cctrl->targtstl_y);
+
+    if (NULL != target)
```

Comment:
The KeeperFX code generally does this the other way around. So `target != NULL`. Not just here but in different places in your code. Would be nice if you could keep this consistent with the rest of the code.

Reply:
Done.